keps/sig-node/3570-cpumanager/README.md

```diff
@@ -323,6 +407,257 @@ func (p *staticPolicy) RemoveContainer(s State, containerID string) error {
+
+Yes, covered by e2e tests
+
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+A rollout can fail if a bug in the cpumanager prevents _new_ pods to start, or existing pods to be restarted.
+Already running workload will not be affected if the node state is steady
+
+###### What specific metrics should inform a rollback?
+
+"cpu_manager_pinning_errors_total". It must be noted that even in fully healthy system there are known benign condition
+that can cause CPU allocation failures. Few selected examples are:
+
+- requesting odd numbered cores (not a full physical core) when the cpumanager is configured with the `full-pcpus-only` option
+- requesting NUMA-aligned cores, with Topology Manager enabled.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+No to both.
+Changes in behavior only affects pods meeting the conditions (guaranteed QoS, integral CPU request) scheduler after the upgrade.
+Running pods will be unaffected by any change. This offers some degree of safety in both upgrade->rollback
+and upgrade->downgrade->upgrade scenarios.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No
+
+### Monitoring Requirements
+
+Monitor the metrics
+- "cpu_manager_pinning_requests_total"
+- "cpu_manager_pinning_errors_total"
+
+###### How can an operator determine if the feature is in use by workloads?
+
+In order for pods to request exclusive CPUs allocation and pinning, they need to match
+all the following criteria:
+- the pod QoS must be "guaranteed"
+- the resources request of CPU (`pod.spec.containers[].resources.limits.cpu`) must be integral.
+
+On top of that, at kubelet level
+- the cpumanager policy must be `static`.
+
+If all the criteria are met, then the feature is in use by workloads.
+
+###### How can someone using this feature know that it is working for their instance?
+
+
+- [X] Other (treat as last resort)
+  - Details: check the kubelet metric `cpu_manager_pinning_requests_total`
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+"cpu_manager_pinning_requests_total" and "cpu_manager_pinning_errors_total"
+We need to find a careful balance here because we don't want to leak hardware details, or in general informations
+dependent on the worker node hardware configuration (example, even if arguable extreme, is the processor core layout).
+
+It is possible to infer which pod would trigger a CPU pinning from the
+[pod resources request](https://kubernetes.io/docs/tasks/administer-cluster/cpu-management-policies/#static-policy)
+but adding these two metrics is both very cheap and helping for the observability of the system.
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
```

Comment:
We should mention the metrics here instead of selecting `Other`:
```
- [X] Metrics
  - Metric name:
    - cpu_manager_pinning_requests_total
    - cpu_manager_pinning_errors_total
```

Reply:
thanks, changing.